src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -543,44 +543,97 @@ func (st *StorageMinerActorState_I) _isSealVerificationCorrect(rt Runtime, onCha
 	return false // TODO: finish
 }
 
-func (st *StorageMinerActorState_I) _sectorExists(rt Runtime, sectorNo sector.SectorNumber) bool {
+func (st *StorageMinerActorState_I) _sectorExists(sectorNo sector.SectorNumber) bool {
 	_, found := st.Sectors()[sectorNo]
 	return found
 }
 
-// Currently deals must be posted on chain via sma.PublishStorageDeals before CommitSector
-// TODO(optimization): CommitSector could contain a list of deals that are not published yet.
-func (a *StorageMinerActorCode_I) CommitSector(rt Runtime, onChainInfo sector.OnChainSealVerifyInfo) {
+// Deals must be posted on chain via sma.PublishStorageDeals before PreCommitSector
+// TODO(optimization): PreCommitSector could contain a list of deals that are not published yet.
```

Comment:
As far as I can tell, the deal IDs aren't being passed to `PreCommitSector` at all, but they should be.

Reply:
Nevermind, I see it hiding below the comment above.

Reply:
see `sealing.SectorPreCommitInfo` which contains `DealIDs` as an input to `PreCommitSector`